src/DotNetWorker.Core/Hosting/AutoGeneratedClassAttribute.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Identifies class generated by Azure functions worker SDK.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class AutoGeneratedClassAttribute : Attribute
```

Comment:
@jviau  I originally named this as F`unctionsWorkerSdkGeneratedClassAttribute`, but later felt that it is somewhat redundant with the namespace. So switched to Auto prefix. (Same for the method attribute as well).  Feel free to suggest a better name.

Reply:
I am thinking instead of naming based on it being primarily used for auto-generated classes, name based on what it does: auto startup hook. Maybe `HostBuilderStartupHookAttribute`. 

Also to my point for using a new interface vs attributes. The attribute route requires knowledge of using an attribute in _two_ places. The interface route requires single knowledge: implement the interface.